node/src/components/block_executor.rs

```diff
@@ -530,9 +530,9 @@ impl<REv: ReactorEventT> Component<REv> for BlockExecutor {
                         )
                     })
             }
-            Event::BlockAlreadyExists(block) => {
-                effect_builder.handle_linear_chain_block(*block).ignore()
-            }
+            Event::BlockAlreadyExists(block) => effect_builder
+                .announce_block_already_executed(*block)
+                .ignore(),
```

Comment:
Will we remove that once we move the `EraSupervisor` out of `Joiner`?

Reply:
I'm not sure - I think it might still be needed for signalling progress to the `LinearChainSync` (although it can probably be removed when we switch to fast sync).